include/soralog/event.hpp

```diff
@@ -57,12 +57,33 @@ namespace soralog {
           break;
       }
 
+      struct {
+        using iterator_category = std::random_access_iterator_tag;
+        using value_type = char;
+        using reference = value_type &;
+        using pointer = value_type *;
+        using difference_type = ptrdiff_t;
+
+        value_type *pos;
+
+        value_type &operator*() const {
+          return *pos;
+        }
+        constexpr auto &operator++() {
+          pos++;
+          return *this;
+        }
+        constexpr auto &operator++(int n) {
```

Comment:
Is this `it++`?
```suggestion
        constexpr auto &operator++(int) {
```
Or is it `it += n`
```suggestion
        constexpr auto &operator+=(int n) {
```

Reply:
it's correct

Reply:
seems like it should be +=